spring-cloud-task-jdbchdfs-common/src/main/java/org/springframework/cloud/task/jdbchdfs/common/JdbcHdfsConfiguration.java

```diff
@@ -0,0 +1,235 @@
+	private ConfigurableApplicationContext context;
+
+	@Value("${jdbc.hdfs.task.resource:org.springframework.cloud.task.app:"
+			+ "jdbc-hdfs-local-task:1.0.0.BUILD-SNAPSHOT}")
+	private String taskResource;
+
+	@Bean
+	@StepScope
+	public NamedColumnJdbcItemReader namedColumnJdbcItemReader(
+			@Value("#{stepExecutionContext['partClause']}") String partClause) throws Exception {
+		NamedColumnJdbcItemReaderFactory namedColumnJdbcItemReaderFactory = new NamedColumnJdbcItemReaderFactory();
+		namedColumnJdbcItemReaderFactory.setDataSource(dataSource);
+		namedColumnJdbcItemReaderFactory.setPartitionClause(partClause);
+		namedColumnJdbcItemReaderFactory.setTableName(props.getTableName());
+		namedColumnJdbcItemReaderFactory.setColumnNames(props.getColumnNames());
+		namedColumnJdbcItemReaderFactory.setSql(props.getSql());
+		namedColumnJdbcItemReaderFactory.setFetchSize(props.getCommitInterval());
+		namedColumnJdbcItemReaderFactory.setDelimiter(props.getDelimiter());
+		namedColumnJdbcItemReaderFactory.afterPropertiesSet();
+		return namedColumnJdbcItemReaderFactory.getObject();
+	}
+
+	@Bean
+	public JobExplorerFactoryBean jobExplorer() {
+		JobExplorerFactoryBean jobExplorerFactoryBean = new JobExplorerFactoryBean();
+		jobExplorerFactoryBean.setDataSource(this.dataSource);
+		return jobExplorerFactoryBean;
+	}
+
+	@Bean
+	public DeployerPartitionHandler partitionHandler(TaskLauncher taskLauncher, JobExplorer jobExplorer) throws Exception {
+		MavenResource resource = MavenResource.parse(taskResource);
+
+		DeployerPartitionHandler partitionHandler = new DeployerPartitionHandler(taskLauncher, jobExplorer, resource, "workerStep");
+
+		Map<String, String> environmentProperties = new HashMap<>();
+		environmentProperties.put("spring.profiles.active", "worker");
+
+		partitionHandler.setEnvironmentProperties(environmentProperties);
+		partitionHandler.setMaxWorkers(2);
+
+		return partitionHandler;
+	}
+
+	@Bean
+	@StepScope
+	public IncrementalColumnRangePartitioner partitioner(JobExplorer jobExplorer, @Value("#{stepExecutionContext['overrideCheckColumnValue']}") Long overrideValue) {
+		IncrementalColumnRangePartitioner partitioner = new IncrementalColumnRangePartitioner();
+		partitioner.setTable(props.getTableName());
+		partitioner.setColumn(props.getPartitionColumn());
+		partitioner.setJobExplorer(jobExplorer);
+		partitioner.setCheckColumn(props.getCheckColumn());
+		partitioner.setOverrideValue(overrideValue);
+		partitioner.setDataSource(dataSource);
+		partitioner.setPartitions(props.getPartitions());
+		return partitioner;
+	}
+
+	@Bean
+	@Profile("worker")
+	public DeployerStepExecutionHandler stepExecutionHandler(JobExplorer jobExplorer) {
+		DeployerStepExecutionHandler handler = new DeployerStepExecutionHandler(this.context, jobExplorer, this.jobRepository);
+
+		return handler;
+	}
+
+	@Bean
+	@StepScope
+	public HdfsTextItemWriter writer(@Value("#{stepExecutionContext['partSuffix']}") String suffix) throws Exception {
```

Comment:
Just thinking...I know the sample partitioned job in SCT only defines a couple beans as master/worker.  We may want to be more aggressive here.